install-deps.sh

```diff
@@ -48,7 +48,7 @@ function munge_ceph_spec_in {
 function ensure_decent_gcc_on_ubuntu {
     # point gcc to the one offered by g++-7 if the used one is not
     # new enough
-    local old=$(gcc -dumpversion)
+    local old=$(gcc -dumpfullversion -dumpversion)
```

Comment:
please note `-dumpfullversion` is a new option introduced by GCC-7. but we need a solution which supports GCC 5.

Reply:
@tchaikov It's a **tricky** here to add "-dumpfullversion" after gcc whose version is lower than gcc 5:
[root@bdpe833n1 ~]# cat /etc/os-release 
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
[root@bdpe833n1 ~]# **gcc -dumpversion**
4.8.5
[root@bdpe833n1 ~]# **gcc -dumpfullversion -dumpversion**
4.8.5
[root@bdpe833n1 ~]# **gcc -dumpfullversion**
gcc: fatal error: no input files
compilation terminated.
[root@bdpe833n1 ~]#

Reply:
@tchaikov Could we keep using "gcc -dumpfullversion -dumpversion" here?

Reply:
sorry for the confusion, i crossed out my comment after i realized that we need both options for backward compatibility with older GCC.

yes, we should use `gcc -dumpfullversion -dumpversion` here. and might want to error out if codename is not trusty or xenial. as per `gcc(1)`

>        -dumpversion
>            Print the compiler version (for example, 3.0, 6.3.0 or 7)---and don't do anything else.  This is the compiler version used in filesystem paths, specs, can be depending on how the compiler has been configured just a single number (major version), two
>            numbers separated by dot (major and minor version) or three numbers separated by dots (major, minor and patchlevel version).
> 
>        -dumpfullversion
>            Print the full compiler version, always 3 numbers separated by dots, major, minor and patchlevel version.